lib/pure/hashes.nim

```diff
@@ -103,7 +144,14 @@ proc hash*(x: pointer): Hash {.inline.} =
       }
     """
   else:
-    result = cast[Hash](cast[uint](x) shr 3) # skip the alignment
+    #[
+    3 bug fixes:
+    * s/cast[Hash]()/hash()/ (#11764)
+    * s/uint/BiggestInt/
+    * s/x shr 3/x/ (no reason to skip the alignment)
```

Comment:
Comments are not commit messages

Reply:
sure but I wanted to show rationale for the new formula; i've now removed these comments for the most part

Reply:
I understand. But rationale can be explained in the commit messages or PR description : )
Better than get the code all messy, since these are not really explanations but seds